src/model.cpp

```diff
@@ -5,21 +5,31 @@
 #include "special_chars.h"
 #include <array>
 #include <string>
+#include <vector>
+#include <algorithm>
 
 void Model::defineCell(const double& grid_step, const int& max_depth){
   cell = Space(atoms, grid_step, max_depth);
   return;
 }
 
-void Model::storeAtomsInTree(){
-  atomtree = AtomTree(atoms);
+void Model::setAtomListForCalculation(std::vector<std::string> included_elements){
+  atoms.clear();
+  for(int i = 0; i < raw_atom_coordinates.size(); i++){
+    if(std::find(included_elements.begin(), included_elements.end(), std::get<0>(raw_atom_coordinates[i])) != included_elements.end()){

```

Comment:
This line is very hard to read. Please make an auxiliary function with the `inline` keyword, and call it something like "isIncluded()"

```c++
inline bool isIncluded(element_symbol) {
  return std::find(included_elements.begin(), included_elements.end(), element_symbol) != included_elements.end())
}
```

Reply:
When you write "auxiliary function", do you mean a subfunction within the `setAtomListForCalculation` function or a separate function in model.cpp ?

Reply:
I mean a function that is only declared in the `model.cpp` file. Put the declaration of the function at the very top of the file, and put the definition at the very bottom. Check `model_filereading.cpp` to see what I mean.